src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/ConfirmOrganizationUserCommand.cs

```diff
@@ -280,6 +280,6 @@ private async Task HandleConfirmationSideEffectsAsync(Guid organizationId, IEnum
         }
 
         var organizationUserIds = confirmedOrganizationUsers.Select(u => u.Id).ToList();
-        await _collectionRepository.CreateDefaultCollectionsAsync(organizationId, organizationUserIds, defaultUserCollectionName);
+        await _collectionRepository.UpsertDefaultCollectionsAsync(organizationId, organizationUserIds, defaultUserCollectionName);
```

Comment:
This isn't related to my PR, but I came across it because I changed the name of my method.

It looks like we're using the bulk creation path for both single-user and multiple-user confirmations. I understand that our best practice is to have the single-user use case go through the bulk interface. Normally, this is more efficient since it reduces the number of database transactions and has no trade-offs.

However, in this case, the bulk implementation uses SqlBulkCopy, which has higher overhead compared to a regular insert, so using it all the time may not be the most optimized solution. That said, I don’t know if this will cause an issue since I can’t test it in production, but I think it’s something we should keep an eye on.

Reply:
So, after this [bug](https://github.com/bitwarden/server/pull/6139#discussion_r2289203090), I strongly advise against calling this method for a single user confirmation, since that increases the risk of locking the CollectionUser.

This is outside the scope of this ticket, so if we want to address it, I can create a separate ticket.

cc @r-tome @Thomas-Avery

Reply:
Cool. Please move all future discussions to the [ticket](https://bitwarden.atlassian.net/browse/PM-25138).